wolfcrypt/test/test.c

```diff
@@ -22328,19 +22337,20 @@ static int ecc_test_buffers(void)
     TEST_SLEEP();
 
     XMEMSET(plain, 0, sizeof(plain));
+    verify = 1;
 
     do {
     #if defined(WOLFSSL_ASYNC_CRYPT)
         ret = wc_AsyncWait(ret, cliKey.asyncDev, WC_ASYNC_FLAG_CALL_AGAIN);
     #endif
         if (ret == 0)
-            ret = wc_ecc_verify_hash(out, x, plain, sizeof(plain), &verify,
+            ret = wc_ecc_verify_hash(out, x, plain, inLen, &verify,
                 cliKey);
     } while (ret == WC_PENDING_E);
     if (ret < 0)
         ERROR_OUT(-10430, done);
 
-    if (XMEMCMP(plain, in, (word32)ret))
+    if (verify != 0)
         ERROR_OUT(-10431, done);
```

Comment:
@dgarske I want to call  attention to this change in the test. I'm still not completely happy that this test is adding much value and serving the original purpose.

Let me know if you have any additional insight.

Reply:
This for fixing this test. It should be using `in` and `inLen`, not plain. This test should be checking the verify success.